### PR TITLE
Hack SymInt.__iadd__ to be working.

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #94136

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

### CODE REVIEW

File: torch/__init__.py [hunk 1]
```diff
@@ -243,6 +243,11 @@ def __bool__(self):
     def __int__(self):
         return self.node.int_()
 
+    # This is a hack, shouldn't be necessary.  Helps
+    # pyhpc_turbulent_kinetic_energy and vision_maskrcnn
+    def __iadd__(self, other):
+        return self + other
```
Comment (Skylion007) [inline torch/__init__.py:249]: Are you sure this isn't needed for all the other inplace ops? In fact, it seems like most the binary magic methods are missing from this SymInt shim wrapper (except for comparison ones (gt, lt) etc)

Comment (ezyang) [reply torch/__init__.py:249]: The non inplace methods are metaprogrammed in when you load symbolic_shapes module. The inplace methods in principle would  all be needed, but they should be unnecessary and I think there is a dynamo bug

Comment (Skylion007) [reply torch/__init__.py:249]: @ezyang I meant is this bug affected all other inplace magic methods too and what should we do to fix it.

Comment (ezyang) [reply torch/__init__.py:249]: It does but I would rather fix Dynamo to correctly do inplace logic, as it's not supposed to be necessary to define all these

**Review by Skylion007 — APPROVED** [review]

**Pull request closed without merging**